### PR TITLE
Add: BundleAnalyzer, Jest, React-testing-library. Create few example tests

### PR DESCRIPTION
**Pull request by Sydnes2003**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,18 +28,26 @@
...
+    "@testing-library/dom": "^10.4.0",
+    "@testing-library/jest-dom": "^6.5.0",
+    "@testing-library/react": "^16.0.1",
+    "@types/jest": "^29.5.13",
     "@types/node": "^22.5.5",
     "@types/react": "^18.3.3",
     "@types/react-dom": "^18.3.0",
     "@vitejs/plugin-react": "^4.3.1",
-    "eslint": "^9.11.0",
+    "eslint": "^8.57.0",
```
Comment (RenZorRUS) [inline package.json:39]: Please, explain why you downgraded the ESLint version?

File: src/widgets/Header/ui/Header.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ interface HeaderProps {
 
 export const Header: FC<HeaderProps> = ({ className = "" }) => {
     return (
-        <div className={css.headerWrapper}>
+        <div data-testid="Header" className={css.headerWrapper}>
```
Comment (RenZorRUS) [inline src/widgets/Header/ui/Header.tsx:15]: Please explain why you added property: `data-testid`?

File: src/widgets/ProfileLink/ui/ProfileLink.tsx [hunk 3]
```diff
@@ -19,14 +18,15 @@ export const ProfileLink: FC<ProfileLinkProps> = ({ className = "" }) => {
 
     return (
         <Button
+            data-testid="ProfileLink"
```
Comment (RenZorRUS) [inline src/widgets/ProfileLink/ui/ProfileLink.tsx:21]: The same

File: package.json (hunk 1 above)
Comment (Sydnes2003) [reply package.json:39]: I've had a problem with this version of ESLint. Other users reported that [downgrading fixed the problem](https://github.com/vercel/next.js/issues/64409)

File: src/widgets/ProfileLink/ui/ProfileLink.tsx (hunk 3 above)
Comment (Sydnes2003) [reply src/widgets/ProfileLink/ui/ProfileLink.tsx:21]: That is used in react-testing-library method .getByTestId for selecting the component by it's data attribute

File: src/widgets/Header/ui/Header.tsx (hunk 2 above)
Comment (Sydnes2003) [reply src/widgets/Header/ui/Header.tsx:15]: https://testing-library.com/docs/queries/bytestid/

File: src/widgets/ProfileLink/ui/ProfileLink.tsx (hunk 3 above)
Comment (Sydnes2003) [reply src/widgets/ProfileLink/ui/ProfileLink.tsx:21]: https://testing-library.com/docs/queries/bytestid/

**Review by RenZorRUS — APPROVED** [review]

**Pull request merged**